konveyor/mig-controller
# Update migration to support rollback on demand

Updating Itineraries and related code to allow rollback control based on MigMigration CR spec.

Fixes https://github.com/konveyor/mig-controller/issues/663

Related to https://github.com/konveyor/mig-operator/pull/473
Related to https://github.com/konveyor/openshift-velero-plugin/pull/58

## QA steps

Assuming working OCP3 (source cluster) and OCP4 (destination cluster) environments with MTC >= 1.4 and some application on the source cluster.

* Create a MigPlan for an application/namespace which should be migrated
* Execute the migration (via UI or creating migmigration CR), wait until it finishes
* Check destination cluster in the choosen namespace for the migrated application objects, it should be present
* Execute migration rollback (via UI when available or creating migmigration CR with spec.rollback: true), wait until it finishes
* Check the app's namespace on the destination cluster, objects should be removed
* Check the app's namespace on the source cluster, objects should be present and app should be working (unquiesced)
* Execute migration rollback again (via UI when available or creating migmigration CR with spec.rollback: true), wait until it finishes
* Check destination cluster in the choosen namespace for the migrated application objects, it should be present and app working

Example usage of the new migration on demand mechanism - https://youtu.be/T1hAY9ObAG0
